packages/grid/_modules_/grid/hooks/features/rows/useRows.ts

```diff
@@ -2,36 +2,48 @@ import * as React from 'react';
 import { ROWS_CLEARED, ROWS_SET, ROWS_UPDATED } from '../../../constants/eventsConstants';
 import { ApiRef } from '../../../models/api/apiRef';
 import { RowApi } from '../../../models/api/rowApi';
-import { checkRowHasId, RowModel, RowId, RowsProp } from '../../../models/rows';
+import {
+  checkRowHasId,
+  RowModel,
+  RowId,
+  RowsProp,
+  RowIdGetter,
+  RowData,
+} from '../../../models/rows';
 import { useApiMethod } from '../../root/useApiMethod';
 import { useLogger } from '../../utils/useLogger';
 import { useGridState } from '../core/useGridState';
-import { InternalRowsState } from './rowsState';
-
-export function convertRowsPropToState({
-  rows,
-  totalRowCount,
-}: {
-  rows: RowsProp;
-  totalRowCount?: number;
-}): InternalRowsState {
+import { getInitialRowState, InternalRowsState } from './rowsState';
+
+export function addRowId(getRowId: RowIdGetter, rowData: RowData): RowModel {
+  return { ...rowData, id: getRowId(rowData) };
```

Comment:
How about bypassing the whole operation if `getRowId` is no provided?

```suggestion
  if (getRowId) {
    return { ...rowData, id: getRowId(rowData) };
  } else {
    return rowData;
  }
```

Actually, why do we even clone the structure now while we weren't doing it before?

Reply:
Well I'm not sure we want to have different behaviour for when you pass the prop or not.
That would bring edge cases and potential issues one way or the other. I'm more in favour of treating every case the same way 🤔